### ISSUE TITLE
ur_gazebo fails under Kinetic with "Action client not connected: /follow_joint_trajectory"

### ISSUE THREAD

**Issue by SimonBirrell** [ros-industrial/universal_robot#281]
Hi,

I've compiled universal_robot under Kinetic, making the modification mentioned here:
https://github.com/ros-industrial/universal_robot/issues/240

I'm attempting to run the Gazebo demo:
http://wiki.ros.org/ur_gazebo

It sort of works - Gazebo and Rviz come up but execution of any plan fails. The problem seems to be when I run 

```
roslaunch ur10_moveit_config ur10_moveit_planning_execution.launch sim:=true
```

This tries unsuccessfully to connect twice to /follow_joint_tractory then fails with 

```    
Action client not connected: /follow_joint_trajectory
```

This topic doesn't appear with rostopic list.

The output from the first terminal is

```
roslaunch ur_gazebo ur10.launch
... logging to /home/simonbirrell/.ros/log/2de0487e-d0c7-11e6-88db-000c294ec09d/roslaunch-ubuntu-43530.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

xacro: Traditional processing is deprecated. Switch to --inorder processing!
To check for compatibility of your document, use option --check-order.
For more infos, see http://wiki.ros.org/xacro#Processing_Order
started roslaunch server http://ubuntu:33865/

SUMMARY
========

PARAMETERS
 * /arm_controller/action_monitor_rate: 10
 * /arm_controller/constraints/elbow_joint/goal: 0.1
 * /arm_controller/constraints/elbow_joint/trajectory: 0.1
 * /arm_controller/constraints/goal_time: 0.6
 * /arm_controller/constraints/shoulder_lift_joint/goal: 0.1
 * /arm_controller/constraints/shoulder_lift_joint/trajectory: 0.1
 * /arm_controller/constraints/shoulder_pan_joint/goal: 0.1
 * /arm_controller/constraints/shoulder_pan_joint/trajectory: 0.1
 * /arm_controller/constraints/stopped_velocity_tolerance: 0.05
 * /arm_controller/constraints/wrist_1_joint/goal: 0.1
 * /arm_controller/constraints/wrist_1_joint/trajectory: 0.1
 * /arm_controller/constraints/wrist_2_joint/goal: 0.1
 * /arm_controller/constraints/wrist_2_joint/trajectory: 0.1
 * /arm_controller/constraints/wrist_3_joint/goal: 0.1
 * /arm_controller/constraints/wrist_3_joint/trajectory: 0.1
 * /arm_controller/joints: ['shoulder_pan_jo...
 * /arm_controller/state_publish_rate: 25
 * /arm_controller/stop_trajectory_duration: 0.5
 * /arm_controller/type: position_controll...
 * /joint_state_controller/publish_rate: 50
 * /joint_state_controller/type: joint_state_contr...
 * /robot_description: <?xml version="1....
 * /robot_state_publisher/publish_frequency: 50.0
 * /robot_state_publisher/tf_prefix: 
 * /rosdistro: kinetic
 * /rosversion: 1.12.6
 * /use_sim_time: True

NODES
  /
    arm_controller_spawner (controller_manager/controller_manager)
    fake_joint_calibration (rostopic/rostopic)
    gazebo (gazebo_ros/gzserver)
    gazebo_gui (gazebo_ros/gzclient)
    joint_state_controller_spawner (controller_manager/controller_manager)
    robot_state_publisher (robot_state_publisher/robot_state_publisher)
    spawn_gazebo_model (gazebo_ros/spawn_model)

auto-starting new master
process[master]: started with pid [43548]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 2de0487e-d0c7-11e6-88db-000c294ec09d
process[rosout-1]: started with pid [43563]
started core service [/rosout]
process[gazebo-2]: started with pid [43584]
process[gazebo_gui-3]: started with pid [43592]
process[spawn_gazebo_model-4]: started with pid [43597]
process[robot_state_publisher-5]: started with pid [43598]
process[fake_joint_calibration-6]: started with pid [43599]
process[joint_state_controller_spawner-7]: started with pid [43600]
process[arm_controller_spawner-8]: started with pid [43609]
spawn_model script started
[INFO] [1483346886.817248, 0.000000]: Loading model xml from ros parameter
[INFO] [1483346886.895042, 0.000000]: Waiting for service /gazebo/spawn_urdf_model
[ INFO] [1483346887.345580699]: Finished loading Gazebo ROS API Plugin.
[ INFO] [1483346887.364626103]: waitForService: Service [/gazebo/set_physics_properties] has not been advertised, waiting...
[INFO] [1483346891.677425, 0.000000]: Calling service /gazebo/spawn_urdf_model
[INFO] [1483346893.613168, 0.001000]: Spawn status: SpawnModel: Successfully spawned model
[ INFO] [1483346893.886072416, 0.021000000]: waitForService: Service [/gazebo/set_physics_properties] is now available.
[ INFO] [1483346894.318623581, 0.097000000]: Physics dynamic reconfigure ready.
[spawn_gazebo_model-4] process has finished cleanly
log file: /home/simonbirrell/.ros/log/2de0487e-d0c7-11e6-88db-000c294ec09d/spawn_gazebo_model-4*.log
```

and the output from the second terminal is 

```
roslaunch ur10_moveit_config ur10_moveit_planning_execution.launch sim:=true
... logging to /home/simonbirrell/.ros/log/2de0487e-d0c7-11e6-88db-000c294ec09d/roslaunch-ubuntu-44457.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://ubuntu:38343/

SUMMARY
========

PARAMETERS
 * /move_group/allow_trajectory_execution: True
 * /move_group/allowed_execution_duration_scaling: 1.2
 * /move_group/allowed_goal_duration_margin: 0.5
 * /move_group/capabilities: move_group/MoveGr...
 * /move_group/controller_list: [{'action_ns': 'f...
 * /move_group/endeffector/planner_configs: ['SBLkConfigDefau...
 * /move_group/jiggle_fraction: 0.05
 * /move_group/manipulator/longest_valid_segment_fraction: 0.05
 * /move_group/manipulator/planner_configs: ['SBLkConfigDefau...
 * /move_group/max_range: 5.0
 * /move_group/max_safe_path_cost: 1
 * /move_group/moveit_controller_manager: moveit_simple_con...
 * /move_group/moveit_manage_controllers: True
 * /move_group/octomap_resolution: 0.025
 * /move_group/planner_configs/BKPIECEkConfigDefault/border_fraction: 0.9
 * /move_group/planner_configs/BKPIECEkConfigDefault/failed_expansion_score_factor: 0.5
 * /move_group/planner_configs/BKPIECEkConfigDefault/min_valid_path_fraction: 0.5
 * /move_group/planner_configs/BKPIECEkConfigDefault/range: 0.0
 * /move_group/planner_configs/BKPIECEkConfigDefault/type: geometric::BKPIECE
 * /move_group/planner_configs/ESTkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/ESTkConfigDefault/range: 0.0
 * /move_group/planner_configs/ESTkConfigDefault/type: geometric::EST
 * /move_group/planner_configs/KPIECEkConfigDefault/border_fraction: 0.9
 * /move_group/planner_configs/KPIECEkConfigDefault/failed_expansion_score_factor: 0.5
 * /move_group/planner_configs/KPIECEkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/KPIECEkConfigDefault/min_valid_path_fraction: 0.5
 * /move_group/planner_configs/KPIECEkConfigDefault/range: 0.0
 * /move_group/planner_configs/KPIECEkConfigDefault/type: geometric::KPIECE
 * /move_group/planner_configs/LBKPIECEkConfigDefault/border_fraction: 0.9
 * /move_group/planner_configs/LBKPIECEkConfigDefault/min_valid_path_fraction: 0.5
 * /move_group/planner_configs/LBKPIECEkConfigDefault/range: 0.0
 * /move_group/planner_configs/LBKPIECEkConfigDefault/type: geometric::LBKPIECE
 * /move_group/planner_configs/PRMkConfigDefault/max_nearest_neighbors: 10
 * /move_group/planner_configs/PRMkConfigDefault/type: geometric::PRM
 * /move_group/planner_configs/PRMstarkConfigDefault/type: geometric::PRMstar
 * /move_group/planner_configs/RRTConnectkConfigDefault/range: 0.0
 * /move_group/planner_configs/RRTConnectkConfigDefault/type: geometric::RRTCon...
 * /move_group/planner_configs/RRTkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/RRTkConfigDefault/range: 0.0
 * /move_group/planner_configs/RRTkConfigDefault/type: geometric::RRT
 * /move_group/planner_configs/RRTstarkConfigDefault/delay_collision_checking: 1
 * /move_group/planner_configs/RRTstarkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/RRTstarkConfigDefault/range: 0.0
 * /move_group/planner_configs/RRTstarkConfigDefault/type: geometric::RRTstar
 * /move_group/planner_configs/SBLkConfigDefault/range: 0.0
 * /move_group/planner_configs/SBLkConfigDefault/type: geometric::SBL
 * /move_group/planner_configs/TRRTkConfigDefault/frountierNodeRatio: 0.1
 * /move_group/planner_configs/TRRTkConfigDefault/frountier_threshold: 0.0
 * /move_group/planner_configs/TRRTkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/TRRTkConfigDefault/init_temperature: 10e-6
 * /move_group/planner_configs/TRRTkConfigDefault/k_constant: 0.0
 * /move_group/planner_configs/TRRTkConfigDefault/max_states_failed: 10
 * /move_group/planner_configs/TRRTkConfigDefault/min_temperature: 10e-10
 * /move_group/planner_configs/TRRTkConfigDefault/range: 0.0
 * /move_group/planner_configs/TRRTkConfigDefault/temp_change_factor: 2.0
 * /move_group/planner_configs/TRRTkConfigDefault/type: geometric::TRRT
 * /move_group/planning_plugin: ompl_interface/OM...
 * /move_group/planning_scene_monitor/publish_geometry_updates: True
 * /move_group/planning_scene_monitor/publish_planning_scene: True
 * /move_group/planning_scene_monitor/publish_state_updates: True
 * /move_group/planning_scene_monitor/publish_transforms_updates: True
 * /move_group/request_adapters: default_planner_r...
 * /move_group/start_state_max_bounds_error: 0.1
 * /move_group/trajectory_execution/execution_duration_monitoring: False
 * /move_group/use_controller_manager: False
 * /robot_description_kinematics/manipulator/kinematics_solver: kdl_kinematics_pl...
 * /robot_description_kinematics/manipulator/kinematics_solver_attempts: 3
 * /robot_description_kinematics/manipulator/kinematics_solver_search_resolution: 0.005
 * /robot_description_kinematics/manipulator/kinematics_solver_timeout: 0.005
 * /robot_description_planning/joint_limits/elbow_joint/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/elbow_joint/has_velocity_limits: True
 * /robot_description_planning/joint_limits/elbow_joint/max_acceleration: 3.15
 * /robot_description_planning/joint_limits/elbow_joint/max_velocity: 3.15
 * /robot_description_planning/joint_limits/shoulder_lift_joint/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/shoulder_lift_joint/has_velocity_limits: True
 * /robot_description_planning/joint_limits/shoulder_lift_joint/max_acceleration: 2.16
 * /robot_description_planning/joint_limits/shoulder_lift_joint/max_velocity: 2.16
 * /robot_description_planning/joint_limits/shoulder_pan_joint/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/shoulder_pan_joint/has_velocity_limits: True
 * /robot_description_planning/joint_limits/shoulder_pan_joint/max_acceleration: 2.16
 * /robot_description_planning/joint_limits/shoulder_pan_joint/max_velocity: 2.16
 * /robot_description_planning/joint_limits/wrist_1_joint/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/wrist_1_joint/has_velocity_limits: True
 * /robot_description_planning/joint_limits/wrist_1_joint/max_acceleration: 3.2
 * /robot_description_planning/joint_limits/wrist_1_joint/max_velocity: 3.2
 * /robot_description_planning/joint_limits/wrist_2_joint/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/wrist_2_joint/has_velocity_limits: True
 * /robot_description_planning/joint_limits/wrist_2_joint/max_acceleration: 3.2
 * /robot_description_planning/joint_limits/wrist_2_joint/max_velocity: 3.2
 * /robot_description_planning/joint_limits/wrist_3_joint/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/wrist_3_joint/has_velocity_limits: True
 * /robot_description_planning/joint_limits/wrist_3_joint/max_acceleration: 3.2
 * /robot_description_planning/joint_limits/wrist_3_joint/max_velocity: 3.2
 * /robot_description_semantic: <?xml version="1....
 * /rosdistro: kinetic
 * /rosversion: 1.12.6

NODES
  /
    move_group (moveit_ros_move_group/move_group)

ROS_MASTER_URI=http://localhost:11311

core service [/rosout] found
process[move_group-1]: started with pid [44569]
[ INFO] [1483346950.422490853, 11.970000000]: Loading robot model 'ur10'...
[ WARN] [1483346950.425821981, 11.971000000]: Skipping virtual joint 'fixed_base' because its child frame 'base_link' does not match the URDF frame 'world'
[ INFO] [1483346950.427840205, 11.971000000]: No root/virtual joint specified in SRDF. Assuming fixed joint
[ INFO] [1483346951.216891265, 12.093000000]: Loading robot model 'ur10'...
[ WARN] [1483346951.217405588, 12.093000000]: Skipping virtual joint 'fixed_base' because its child frame 'base_link' does not match the URDF frame 'world'
[ INFO] [1483346951.218931386, 12.094000000]: No root/virtual joint specified in SRDF. Assuming fixed joint
[ INFO] [1483346951.794654711, 12.178000000]: Publishing maintained planning scene on 'monitored_planning_scene'
[ INFO] [1483346951.854467677, 12.184000000]: MoveGroup debug mode is OFF
Starting context monitors...
[ INFO] [1483346951.854583343, 12.184000000]: Starting scene monitor
[ INFO] [1483346951.913018531, 12.190000000]: Listening to '/planning_scene'
[ INFO] [1483346951.913156217, 12.190000000]: Starting world geometry monitor
[ INFO] [1483346951.993966765, 12.197000000]: Listening to '/collision_object' using message notifier with target frame '/world '
[ INFO] [1483346952.049374421, 12.204000000]: Listening to '/planning_scene_world' for planning scene world geometry
[ INFO] [1483346952.223807477, 12.224000000]: Listening to '/attached_collision_object' for attached collision objects
Context monitors started.
[ INFO] [1483346952.978439618, 12.336000000]: Initializing OMPL interface using ROS parameters
[ INFO] [1483346953.359928423, 12.391000000]: Using planning interface 'OMPL'
[ INFO] [1483346953.736397098, 12.443000000]: Param 'default_workspace_bounds' was not set. Using default value: 10
[ INFO] [1483346953.746320202, 12.443000000]: Param 'start_state_max_bounds_error' was set to 0.1
[ INFO] [1483346953.757789694, 12.443000000]: Param 'start_state_max_dt' was not set. Using default value: 0.5
[ INFO] [1483346953.762668038, 12.444000000]: Param 'start_state_max_dt' was not set. Using default value: 0.5
[ INFO] [1483346953.774779030, 12.444000000]: Param 'jiggle_fraction' was set to 0.05
[ INFO] [1483346953.779849285, 12.445000000]: Param 'max_sampling_attempts' was not set. Using default value: 100
[ INFO] [1483346953.782368662, 12.445000000]: Using planning request adapter 'Add Time Parameterization'
[ INFO] [1483346953.782524586, 12.445000000]: Using planning request adapter 'Fix Workspace Bounds'
[ INFO] [1483346953.782727986, 12.445000000]: Using planning request adapter 'Fix Start State Bounds'
[ INFO] [1483346953.782952645, 12.445000000]: Using planning request adapter 'Fix Start State In Collision'
[ INFO] [1483346953.784730850, 12.446000000]: Using planning request adapter 'Fix Start State Path Constraints'
[ WARN] [1483346953.814066036, 12.450000000]: 
Deprecation warning: parameter 'allowed_execution_duration_scaling' moved into namespace 'trajectory_execution'.
Please, adjust file trajectory_execution.launch.xml!
[ WARN] [1483346953.821469889, 12.450000000]: 
Deprecation warning: parameter 'allowed_goal_duration_margin' moved into namespace 'trajectory_execution'.
Please, adjust file trajectory_execution.launch.xml!
[ INFO] [1483346974.874731068, 17.527000000]: Waiting for /follow_joint_trajectory to come up
[ INFO] [1483347002.908186061, 22.527000000]: Waiting for /follow_joint_trajectory to come up
[ERROR] [1483347028.430754822, 27.527000000]: Action client not connected: /follow_joint_trajectory
[ INFO] [1483347028.475887968, 27.534000000]: Returned 0 controllers in list
[ INFO] [1483347028.727797424, 27.560000000]: Trajectory execution is managing controllers
Loading 'move_group/ApplyPlanningSceneService'...
Loading 'move_group/ClearOctomapService'...
Loading 'move_group/MoveGroupCartesianPathService'...
Loading 'move_group/MoveGroupExecuteService'...
Loading 'move_group/MoveGroupExecuteTrajectoryAction'...
Loading 'move_group/MoveGroupGetPlanningSceneService'...
Loading 'move_group/MoveGroupKinematicsService'...
Loading 'move_group/MoveGroupMoveAction'...
Loading 'move_group/MoveGroupPickPlaceAction'...
Loading 'move_group/MoveGroupPlanService'...
Loading 'move_group/MoveGroupQueryPlannersService'...
Loading 'move_group/MoveGroupStateValidationService'...
[ INFO] [1483347031.454296150, 27.853000000]: 

********************************************************
* MoveGroup using: 
*     - ApplyPlanningSceneService
*     - ClearOctomapService
*     - CartesianPathService
*     - ExecuteTrajectoryService
*     - ExecuteTrajectoryAction
*     - GetPlanningSceneService
*     - KinematicsService
*     - MoveAction
*     - PickPlaceAction
*     - MotionPlanService
*     - QueryPlannersService
*     - StateValidationService
********************************************************

[ INFO] [1483347031.454952305, 27.853000000]: MoveGroup context using planning plugin ompl_interface/OMPLPlanner
[ INFO] [1483347031.456931657, 27.853000000]: MoveGroup context initialization complete

You can start planning now!
```

This seems similar to issue 159
https://github.com/ros-industrial/universal_robot/issues/159
and suggests that the action server has gone down for some reason. However, I don't see any error messages in the first terminal.

Any ideas? I know the Kinetic version isn't launched yet but other reasons, I really really need to move to Kinetic. Thanks!

Simon

**Comment by shaun-edwards** [comment]
@SimonBirrell, thanks for reporting.  I was curious, so I tried to launch the gazebo ur sim on my system (in indigo).  I thought I'd at least be able to narrow it down to a ROS distro difference.  Unfortunately, I can't even load the xacro in indigo :(
```
shaun@shaun-Precision-5510:~/ros_industrial/universal/src/universal_robot/ur_description/urdf$ rosrun xacro xacro.py ur5_robot.urdf.xacro 
...
xacro.XacroException: Property wasn't defined: u'pi'
```

Can you determine if the action server has started and/or is running?  Running `rostopic` to find the action topics and `rosnode` would be helpful.

**Comment by SimonBirrell** [comment]
Hi!

```
rosnode list
/fake_joint_calibration
/gazebo
/move_group
/robot_state_publisher
/rosout
```

and

```
rostopic list
/attached_collision_object
/calibrated
/clock
/collision_object
/execute_trajectory/cancel
/execute_trajectory/feedback
/execute_trajectory/goal
/execute_trajectory/result
/execute_trajectory/status
/gazebo/link_states
/gazebo/model_states
/gazebo/parameter_descriptions
/gazebo/parameter_updates
/gazebo/set_link_state
/gazebo/set_model_state
/joint_states
/move_group/cancel
/move_group/display_contacts
/move_group/display_planned_path
/move_group/feedback
/move_group/goal
/move_group/monitored_planning_scene
/move_group/ompl/parameter_descriptions
/move_group/ompl/parameter_updates
/move_group/plan_execution/parameter_descriptions
/move_group/plan_execution/parameter_updates
/move_group/planning_scene_monitor/parameter_descriptions
/move_group/planning_scene_monitor/parameter_updates
/move_group/result
/move_group/sense_for_plan/parameter_descriptions
/move_group/sense_for_plan/parameter_updates
/move_group/status
/move_group/trajectory_execution/parameter_descriptions
/move_group/trajectory_execution/parameter_updates
/pickup/cancel
/pickup/feedback
/pickup/goal
/pickup/result
/pickup/status
/place/cancel
/place/feedback
/place/goal
/place/result
/place/status
/planning_scene
/planning_scene_world
/rosout
/rosout_agg
/tf
/tf_static
/trajectory_execution_event
```

This is just with the first two terminal windows running:

```
roslaunch ur_gazebo ur10.launch
roslaunch ur10_moveit_config ur10_moveit_planning_execution.launch sim:=true
```

Thanks!

Simon

**Comment by shaun-edwards** [comment]
I have a similar sim (not UR, but kuka).  It has a ` /joint_trajectory_action/goal` topic.  When i run `rostopic info`, I get:
```
Type: control_msgs/FollowJointTrajectoryActionGoal

Publishers: 
 * /my_node (http://shaun-Precision-5510:36613/)

Subscribers: 
 * /gazebo (http://shaun-Precision-5510:40339/)
```

I believe the `arm_controller_spawner` node is what loads the action into gazebo.  You should see messages similar to below, when everything is working:

```
[ INFO] [1483378160.855813772, 0.170000000]: Loading gazebo_ros_control plugin
[ INFO] [1483378160.855852267, 0.170000000]: Starting gazebo_ros_control plugin in namespace: /
[ INFO] [1483378160.856910019, 0.170000000]: gazebo_ros_control plugin is waiting for model URDF in parameter [robot_description] on the ROS param server.
[ INFO] [1483378160.997802758, 0.170000000]: Loaded gazebo_ros_control.
[arm_controller_spawner-8] process has finished cleanly
```

**Comment by gavanderhoorn** [comment]
@SimonBirrell: I can't help you with Kinetic, but I at least fixed `ur_description` for Indigo and slightly improved user experience with `ur_gazebo` and `urN_moveit_config`.

Following the instructions in the readme now works for me to start up a Gazebo sim with a UR, MoveIt and RViz with the MoveIt plugin. Planning and execution all work.

**Comment by gavanderhoorn** [comment]
@shaun-edwards: if you're using 'my' Gazebo pkgs, then the setup in those is not directly comparable to the one `ur_gazebo` uses. They're similar, but not identical.

**Comment by shaun-edwards** [comment]
@gavanderhoorn, do you have a PR for the `ur_description` fix?

**Comment by gavanderhoorn** [comment]
No. I used my administrative access to push the fixes directly to `indigo-devel`. The current release is broken and I've a very narrow window to get a fixed version synced with the public repositories.

A brake with policy, but I deemed it acceptable in this situation.

**Comment by gavanderhoorn** [comment]
See #282 for the sync to the release branch `indigo`.

**Comment by shaun-edwards** [comment]
Ok...just making sure I didn't miss something.  Thanks for helping out!

**Comment by gavanderhoorn** [comment]
Released in ros/rosdistro#13516.

**Comment by gavanderhoorn** [comment]
This issue is somewhat of a duplicate of #231, but I've asked the OP of that issue whether we can merge it with this one.

**Comment by gavanderhoorn** [comment]
@SimonBirrell wrote:
> This seems similar to issue #159 and suggests that the action server has gone down for some reason. However, I don't see any error messages in the first terminal.

This is most likely a rather different issue, as `ur_driver` is not involved when running things against Gazebo (or any simulation really).

**Comment by shaun-edwards** [comment]
FYI, the joint trajectory action appears to be running in indigo with @gavanderhoorn's fix:

`rostopic list`
```
/arm_controller/command
/arm_controller/follow_joint_trajectory/cancel
/arm_controller/follow_joint_trajectory/feedback
/arm_controller/follow_joint_trajectory/goal
/arm_controller/follow_joint_trajectory/result
/arm_controller/follow_joint_trajectory/status
/arm_controller/state
```

**Comment by gavanderhoorn** [comment]
There was never really a problem with the action server (it's run by `gazebo_ros_control`): the problem was that a constant used by the xacros was undefined, leading to the `ur_gazebo/urX.launch` file never completely finishing, thus not loading Gazebo or spawning the controllers. That in turn leads to the action interface never coming up, etc etc.

I don't know right now what the problem is under Kinetic, but at least on Indigo that is what happened before `1.1.9` (and since `1.1.7`).

**Comment by SimonBirrell** [comment]
Hi,

I pulled 1.1.9 and did a catkin_make. The problem is still there.

I've noticed a new error message when running roslaunch ur_gazebo ur10.launch

```
roslaunch ur_gazebo ur10.launch
... logging to /home/simonbirrell/.ros/log/f8da7b6c-d18a-11e6-88db-000c294ec09d/roslaunch-ubuntu-68758.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

redefining global property: pi
when processing file: /home/simonbirrell/catkin_ws/src/universal_robot/ur_description/urdf/ur10.urdf.xacro
included from: /home/simonbirrell/catkin_ws/src/universal_robot/ur_description/urdf/ur10_robot.urdf.xacro
started roslaunch server http://ubuntu:44427/

SUMMARY
========

PARAMETERS
 * /arm_controller/action_monitor_rate: 10
 * /arm_controller/constraints/elbow_joint/goal: 0.1
 * /arm_controller/constraints/elbow_joint/trajectory: 0.1
 * /arm_controller/constraints/goal_time: 0.6
 * /arm_controller/constraints/shoulder_lift_joint/goal: 0.1
 * /arm_controller/constraints/shoulder_lift_joint/trajectory: 0.1
 * /arm_controller/constraints/shoulder_pan_joint/goal: 0.1
 * /arm_controller/constraints/shoulder_pan_joint/trajectory: 0.1
 * /arm_controller/constraints/stopped_velocity_tolerance: 0.05
 * /arm_controller/constraints/wrist_1_joint/goal: 0.1
 * /arm_controller/constraints/wrist_1_joint/trajectory: 0.1
 * /arm_controller/constraints/wrist_2_joint/goal: 0.1
 * /arm_controller/constraints/wrist_2_joint/trajectory: 0.1
 * /arm_controller/constraints/wrist_3_joint/goal: 0.1
 * /arm_controller/constraints/wrist_3_joint/trajectory: 0.1
 * /arm_controller/joints: ['shoulder_pan_jo...
 * /arm_controller/state_publish_rate: 25
 * /arm_controller/stop_trajectory_duration: 0.5
 * /arm_controller/type: position_controll...
 * /joint_state_controller/publish_rate: 50
 * /joint_state_controller/type: joint_state_contr...
 * /robot_description: <?xml version="1....
 * /robot_state_publisher/publish_frequency: 50.0
 * /robot_state_publisher/tf_prefix: 
 * /rosdistro: kinetic
 * /rosversion: 1.12.6
 * /use_sim_time: True

NODES
  /
    arm_controller_spawner (controller_manager/controller_manager)
    fake_joint_calibration (rostopic/rostopic)
    gazebo (gazebo_ros/gzserver)
    gazebo_gui (gazebo_ros/gzclient)
    joint_state_controller_spawner (controller_manager/controller_manager)
    robot_state_publisher (robot_state_publisher/robot_state_publisher)
    spawn_gazebo_model (gazebo_ros/spawn_model)

auto-starting new master
process[master]: started with pid [68772]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to f8da7b6c-d18a-11e6-88db-000c294ec09d
process[rosout-1]: started with pid [68785]
started core service [/rosout]
process[gazebo-2]: started with pid [68793]
process[gazebo_gui-3]: started with pid [68804]
process[spawn_gazebo_model-4]: started with pid [68816]
process[robot_state_publisher-5]: started with pid [68820]
process[fake_joint_calibration-6]: started with pid [68821]
process[joint_state_controller_spawner-7]: started with pid [68823]
process[arm_controller_spawner-8]: started with pid [68836]
spawn_model script started
[INFO] [1483430552.497400, 0.000000]: Loading model xml from ros parameter
[INFO] [1483430552.508283, 0.000000]: Waiting for service /gazebo/spawn_urdf_model
[ INFO] [1483430552.687976917]: Finished loading Gazebo ROS API Plugin.
[ INFO] [1483430552.695259646]: waitForService: Service [/gazebo/set_physics_properties] has not been advertised, waiting...
[ INFO] [1483430553.959535151, 0.022000000]: waitForService: Service [/gazebo/set_physics_properties] is now available.
[INFO] [1483430554.030640, 0.071000]: Calling service /gazebo/spawn_urdf_model
[INFO] [1483430554.739896, 0.285000]: Spawn status: SpawnModel: Successfully spawned model
[ INFO] [1483430554.819791086, 0.285000000]: Physics dynamic reconfigure ready.
Unhandled exception in thread started by 
sys.excepthook is missing
lost sys.stderr
[spawn_gazebo_model-4] process has finished cleanly
log file: /home/simonbirrell/.ros/log/f8da7b6c-d18a-11e6-88db-000c294ec09d/spawn_gazebo_model-4*.log
```

It's towards the end:

```
Unhandled exception in thread started by 
sys.excepthook is missing
lost sys.stderr
```

That wasn't there when I opened the issue, but I'm not sure when it started appearing.

It's clear that arm_controller isn't loading for some reason. I'm guessing what to try next, but I've tried:

1. Running roslaunch ur_gazebo ur10.launch in one terminal
2. Launching the arm_controller_spawner manually with:

```
rosrun controller_manager controller_manager spawn arm_controller respawn:="false" output:="screen"
```

I get no output at all, but this is probably the wrong way to go about things.

What can I try next to track down the issue?

Thanks

Simon

**Comment by shaun-edwards** [comment]
Do you know which node is throwing the unhandled exception?  Once fully loaded, you should see the following nodes:
```
shaun@shaun-Precision-5510:~/ros_industrial/universal$ rosnode list
/fake_joint_calibration
/gazebo
/robot_state_publisher
/rosout
```
In addition to the terminal output, you should see the Gazebo sim window, like below:
![image](https://cloud.githubusercontent.com/assets/2932571/21614170/000b26c6-d19e-11e6-824a-19dc2e6aa0b7.png)

**Comment by SimonBirrell** [comment]
Here's what the screen looks like after running 

```
roslaunch ur_gazebo ur10.launch
```

![screenshot 2017-01-04 09 30 42](https://cloud.githubusercontent.com/assets/1873370/21636003/1cfa9f78-d261-11e6-8928-954dab1da065.png)

So I have the same nodes as as you. Today I didn't get the exception. Go figure. I then run the second terminal:

```
roslaunch ur10_moveit_config ur10_moveit_planning_execution.launch sim:=true
```

![screenshot 2017-01-04 09 33 58](https://cloud.githubusercontent.com/assets/1873370/21636045/464d99de-d261-11e6-96ad-4fc11b0d04a3.png)

The same nodes still, no sign of arm_controller.

Simon

**Comment by SimonBirrell** [comment]
Are we aware of anybody who has this working on Kinetic? Or anyone else who has tried?

**Comment by shaun-edwards** [comment]
@SimonBirrell, I know people are using Kinetic, but I'm not aware of anyone who is using the universal gazebo sim.  I recommend and email to the users group (swri-ros-pkg-dev@googlegroups.com).  There a number of users of universal who might be able to help.  Sorry this isn't working for you.

**Comment by SimonBirrell** [comment]
FOUND IT!

The solution is absurdly simple, of course. You need to reinstall the gazebo_ros_pkgs. Presumably the latest version isn't included in the version of Gazebo that comes with Kinetic.

So just type

```
sudo apt-get install ros-kinetic-gazebo-ros-pkgs ros-kinetic-gazebo-ros-control
```

after installing Kinetic and all the bits and pieces listed on http://wiki.ros.org/ur_gazebo and applying fix #240 . 
More details at http://gazebosim.org/tutorials?tut=ros_installing

I'm still having some issues executing paths correctly, but this is clearly a separate issue. All /arm_controller/follow_joint_trajectory topics are now present and correct.

It seems to me you may have a functioning Kinect release once you accept #240. Might be worth taking the plunge.

Thanks for all the help and advice.

**Comment by SimonBirrell** [comment]
A final note for the lost souls who find this. I also needed to adjust the parameter longest_valid_segment_fraction to be 0.02 in ur10_moveit_config/config/ompl_planning.yaml
as detailed in 
https://groups.google.com/forum/#%21topic/moveit-users/zNRG50tHuuY

Substitute ur5 for ur10 as necessary. 

With that, I have a fully-working Gazebo demo of the UR10 under Kinetic!
